web/src/components/PasswordWithMeter.js

```diff
@@ -0,0 +1,105 @@
+import { Choice, TextField } from '@cmsgov/design-system-core';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import zxcvbn from 'zxcvbn';
+
+class Password extends Component {
+  state = { showPassword: false, strength: 0 };
+
+  componentWillMount() {
+    const { value } = this.props;
+    if (value) {
+      this.changePassword({ target: { value } });
+    }
+  }
+
+  changePassword = e => {
+    const { compareTo, onChange } = this.props;
+    this.setState({ strength: zxcvbn(e.target.value, compareTo).score });
+
+    if (onChange) {
+      onChange(e);
+    }
+  };
+
+  toggleShowPassword = () => {
+    this.setState(prev => ({ showPassword: !prev.showPassword }));
+  };
+
+  render() {
+    const { title, value, ...rest } = this.props;
+    const { showPassword, strength } = this.state;
+
+    let passwordQuality = 'Password strength: Weak';
+    if (value.length === 0) {
+      passwordQuality = <span>&nbsp;</span>;
+    } else if (strength === 3) {
+      passwordQuality = 'Password strength: Good';
+    } else if (strength > 3) {
+      passwordQuality = 'Password strength: Great';
+    }
+
+    const strengthClass = [
+      'strength-meter',
+      value.length > 0 ? 'visible' : 'hidden'
```

Comment:
Oh, good catch, that needs to be cleaned up.  The meter originally disappeared while the password field was empty, but Jen pointed out that was a jarring user experience, to have stuff appear and move content down just because you typed something.